### PR TITLE
Add some new seeds and allow harvesting flowers

### PR DESCRIPTION
**Pull request by Rail-Runner**
#### Summary

```SUMMARY: Content "Add some new seeds and allow harvesting flowers"```
#### Purpose of change

Adding new content and improving consistency of things.
1. Add cattail seeds, cactus seeds, dahlia seeds and decorative plant seeds. Cattail, cactus and dahlia produce some useful products if grown and their seeds (except cactus) can be harvested in wilderness, decorative plants only provide "withered plant" items and are mostly useless for growing (but their seeds can, for example, be used for crafting bird food)

2. Most wild flowers that were completely useless before can now be harvested for "decorative plant seeds" and "withered plant" items.

3. Add missing and newly-added seeds to bird food recipe.

4. Rewrite some harvesting code to reduce code duplication and to clarify messages in some places.

5. Make Burrowing mutation a valid substitute for a digging tool when harvesting dahlia roots.
#### Describe the solution

JSON editing. As related to harvesting rewrite, I've written a function to be used by plant harvesting examine functions, to replace the block that is nearly identical and simplified a few checks and calculations for better performance.
#### Describe alternatives you've considered

None.
#### Additional context

According to the map in this Wikipedia article below, some kind(s) of cactuses grow in New England. I didn't add them as wilderness plants, though.
https://en.wikipedia.org/wiki/Cactus#Distribution

### CODE REVIEW

**Comment by RarkGrames** [comment]
Does this mean that any plant that has a "furniture" can be made to drop stuff only by .json editing?

**Comment by Rail-Runner** [comment]
> Does this mean that any plant that has a "furniture" can be made to drop stuff only by .json editing?

Yes, you can add "harvest_by_season" entry to any furniture (or maybe even terrain, but I didn't check that)

**Comment by RarkGrames** [comment]
> 
> 
> > Does this mean that any plant that has a "furniture" can be made to drop stuff only by .json editing?
> 
> Yes, you can add "harvest_by_season" entry to any furniture (or maybe even terrain, but I didn't check that)

Nice. Now my lifelong dream of drinking chamomile tea can become a reality.

File: src/iexamine.cpp [hunk 1]
```diff
@@ -1609,27 +1564,27 @@ void iexamine::flower_dahlia( player &p, const tripoint &examp )
         return;
     }
 
-    if( !p.has_quality( quality_id( "DIG" ) ) ) {
-        none( p, examp );
-        add_msg( m_info, _( "If only you had a shovel to dig up those roots..." ) );
-        return;
-    }
+    bool can_get_root = (p.has_quality( quality_id( "DIG" ) ) || p.has_trait( trait_BURROW ) );
```
Comment (ZhilkinSerg) [inline src/iexamine.cpp]:
```suggestion
    bool can_get_root = p.has_quality( quality_id( "DIG" ) || p.has_trait( trait_BURROW );
```

File: src/iexamine.cpp [hunk 2]
```diff
@@ -1609,27 +1564,27 @@ void iexamine::flower_dahlia( player &p, const tripoint &examp )
...
-        p.can_pickVolume( item( "dahlia_root" ), true ) ) {
-        p.i_add( item( "dahlia_root" ) );
-        p.add_msg_if_player( _( "You harvest: dahlia root" ) );
-    } else {
-        g->m.add_item_or_charges( p.pos(), item( "dahlia_root" ) );
-        p.add_msg_if_player( _( "You harvest and drop: dahlia root" ) );
-    }
+    if ( can_get_root ) {
+        handle_harvest( p, "dahlia_root", false );
+	}
```
Comment (ZhilkinSerg) [inline src/iexamine.cpp]:
```suggestion
    }
```

File: src/iexamine.cpp [hunk 3]
```diff
@@ -1609,27 +1564,27 @@ void iexamine::flower_dahlia( player &p, const tripoint &examp )
...
-        none( p, examp );
-        return;
-    }
+    if( can_get_root ) {
+        if( !query_yn( _( "Pick %s?" ), g->m.furnname( examp ) ) ) {
+            none( p, examp );
+            return;
+        }
+    } else {
+        if( !query_yn( _( "You don't have a digging tool to dig up roots. Pick %s anyway?" ), g->m.furnname( examp ) ) ) {
```
Comment (ZhilkinSerg) [inline src/iexamine.cpp]:
```suggestion
        if( !query_yn( _( "You don't have a digging tool to dig up roots. Pick %s anyway?" ),
                       g->m.furnname( examp ) ) ) {
```

File: src/iexamine.cpp [hunk 4]
```diff
@@ -1609,27 +1564,27 @@ void iexamine::flower_dahlia( player &p, const tripoint &examp )
...
 
-    if( p.can_pickWeight( item( "dahlia_root" ), true ) &&
-        p.can_pickVolume( item( "dahlia_root" ), true ) ) {
-        p.i_add( item( "dahlia_root" ) );
-        p.add_msg_if_player( _( "You harvest: dahlia root" ) );
-    } else {
-        g->m.add_item_or_charges( p.pos(), item( "dahlia_root" ) );
-        p.add_msg_if_player( _( "You harvest and drop: dahlia root" ) );
-    }
+    if ( can_get_root ) {
```
Comment (ZhilkinSerg) [inline src/iexamine.cpp]:
```suggestion
    if( can_get_root ) {
```

File: src/iexamine.cpp [hunk 5]
```diff
@@ -1669,16 +1624,12 @@ static bool harvest_common( player &p, const tripoint &examp, bool furn, bool ne
...
-                g->m.add_item_or_charges( p.pos(), item( entry.drop ) );
-                p.add_msg_if_player( _( "You harvest and drop: %s." ), item( entry.drop ).tname() );
-            }
+        if( roll >= 1 ) {
             got_anything = true;
-        }
+            for( int i = 0; i < roll; i++ ) {
+                handle_harvest( p, entry.drop, false );
+            }
+       }
```
Comment (ZhilkinSerg) [inline src/iexamine.cpp]:
```suggestion
    }
```

**Comment by I-am-Erk** [comment]
I'm not a big fan of a generic seed_withered item... It's not at all clear what it plants, and I'm not sure why not just make item IDs for all the appropriate plant seeds.

**Comment by Rail-Runner** [comment]
> I'm not a big fan of a generic seed_withered item... It's not at all clear what it plants, and I'm not sure why not just make item IDs for all the appropriate plant seeds. Nearly all these plants really do have uses... Someone will have reason to plant them specifically.
> 
> If you don't want to do the legwork of making the different items, you could make a single example one and leave the rest to other contributors. It's a time honoured tradition

I think it complicates things without really effectively changing anything. All those plants have one thing in common, which is that they only drop withered plants when grown and harvested (and this is what actually matters for survivor). If someone ever adds uses specific to any one of those plants, specific seeds, plant harvest products, etc. could be added. Also there are just too many different species of decorative flowers.

**Comment by I-am-Erk** [comment]
Just being able to set up a garden and choose what flowers you plant for roleplay reasons is a solid rationale to keep them separate, I'd say.

Even if kept as a single seed, something like seed_flower would be much better than seed_withered. It's not a withered seed, and seeing that ID in an itemgroup list does a poor job of getting the point across

File: src/iexamine.cpp [hunk 6]
```diff
@@ -1537,63 +1547,8 @@ void iexamine::flower_cactus( player &p, const tripoint &examp )
...
-/**
- * It's a flower, drink nectar if your able to.
- */
-void iexamine::flower_bluebell( player &p, const tripoint &examp )
-{
-    if( dead_plant( true, p, examp ) ) {
-        return;
-    }
-
-    drink_nectar( p );
```
Comment (Night-Pryanik) [inline src/iexamine.cpp]: Why did you remove the ability to drink nectar for bluebells, tulips and spurge?

Comment (Rail-Runner) [reply src/iexamine.cpp]: I didn't remove that, it's still working, just done via another examine function.

To be honest the whole concept of a human-sized creature getting meaningful amounts of nutrition from drinking normal flower nectar (and especially being able to get unlimited amount of nectar from one flower) seems very unrealistic to me, but that's worth a separate discussion.

**Comment by Greevv** [comment]
Prickly pear cacti are native to the NE US. https://www.fs.fed.us/wildflowers/plant-of-the-week/opuntia_humifusa.shtml
